viadot/utils.py

```diff
@@ -103,8 +141,8 @@ def handle_api_response(
     response = get_response(
         url=url, auth=auth, params=params, headers=headers, timeout=timeout
     )
-    response_handled = handle_response(response)
 
+    response_handled = handle_response(response, url)
```

Comment:
Can't you take the url from the Response object? At any rate I don't think you should require another param just to log some details. Use the Response object to get any relevant info or re-raise exceptions.

Reply:
Done